src/ir/stack-utils.cpp

```diff
@@ -164,6 +164,95 @@ bool StackSignature::isSubType(StackSignature a, StackSignature b) {
                     });
 }
 
+bool StackSignature::haveLeastUpperBound(StackSignature a, StackSignature b) {
```

Comment:
Everything in this file and in the stack-utils tests is purely revert from the LUB removal and can be safely ignored.